src/synchronicity/synchronizer.py

```diff
@@ -468,7 +480,7 @@ def _wrap_callable(
         else:
             _name = name
 
-        is_coroutinefunction = inspect.iscoroutinefunction(f)
+        is_coroutinefunction = iscoroutinefunction(f)
```

Comment:
If you make the change I suggested below, this could probably reuse the `is_async` (maybe rename iscoroutinefunction_follow_wrapped?) method there

Reply:
Left this so the equivalence with the generator option is more clear (I don't think we need to avoid unwrapping a synchronicity interface here since we use this to determine whether to do that wrapping, right?)

Reply:
Not 100% sure what makes more sense. If a blocking interface wrapper is attempted to be wrapped again, we might want it to be determined as non-async if it will be called like that. I just tried reuisng the async_wrap variant in synchronizer and it seems to work 🤷